data/secscan_model/secscan_v4_model.py

```diff
@@ -0,0 +1,125 @@
+from math import log10
+from peewee import fn, JOIN
+
+from data.secscan_model.interface import SecurityScannerInterface
+from data.secscan_model.datatypes import ScanLookupStatus, SecurityInformationLookupResult
+from data.registry_model import registry_model
+from util.migrate.allocator import yield_random_entries
+from util.secscan.api import SecurityConfigValidator
+from util.config import URLSchemeAndHostname
+from data import database
+from data.registry_model.datatypes import Manifest
+
+
+class ScanToken(namedtuple("NextScanToken", ["min_id"])):
+    """
+  ScanToken represents an opaque token that can be passed between runs of the security worker
+  to continue scanning whereever the previous run left off. Note that the data of the token is
+  *opaque* to the security worker, and the security worker should *not* pull any data out or modify
+  the token in any way.
+  """
+
+
+class V4SecurityScanner(SecurityScannerInterface):
+    """
+    Implementation of the security scanner interface for Clair V4 API-compatible implementations.
+
+    In addition to consuming a new security scanner API, this class also uses the `ManifestSecurityStatus`
+    database model to track the status of `Manifests`, rather than storing the index status on the `Image`
+    row itself.
+    """
+
+    def __init__(self, app, instance_keys, storage):
+        self.app = app
+        self._legacy_api_handler = None
+
+        validator = SecurityConfigValidator(
+            app.config.get("FEATURE_SECURITY_SCANNER", False),
+            app.config.get("SECURITY_SCANNER_ENDPOINT"),
+        )
+        if not validator.valid():
+            logger.warning("Failed to validate security scanner V2 configuration")
+            return
+        
+        url_scheme_and_hostname = URLSchemeAndHostname(
+            app.config["PREFERRED_URL_SCHEME"], app.config["SERVER_HOSTNAME"]
+        )
+      
+
+    def load_security_information(self, manifest_or_legacy_image, include_vulnerabilities=False):
+        if not isinstance(manifest_or_legacy_image, Manifest):
+            # TODO(alecmerdler): Do we support legacy images?
+            raise NotImplementedError
+
+        status = None
+        try:
+            status = database.ManifestSecurityStatus.get(manifest=manifest_or_legacy_image._db_id)
+        except database.ManifestSecurityStatus.DoesNotExist:
+            return SecurityInformationLookupResult.with_status(
+                ScanLookupStatus.NOT_YET_INDEXED
+            )
+
+        if status.index_status == database.IndexStatus.FAILED:
+            return SecurityInformationLookupResult.with_status(ScanLookupStatus.FAILED_TO_INDEX)
+        if status.index_status == database.IndexStatus.UNSUPPORTED:
+            return SecurityInformationLookupResult.with_status(ScanLookupStatus.UNSUPPORTED_FOR_INDEXING)
+        if status.index_status in [database.IndexStatus.TIMED_OUT, database.IndexStatus.WAITING, database.IndexStatus.IN]:
```

Comment:
`IndexStatus.IN`?

Reply:
Is that a method on `IntEnum`?

Reply:
No idea; you wrote it... not sure what it means :)

Reply:
Oh whoops